src/numerics.jl

```diff
@@ -90,37 +107,79 @@ function N(b::BasicType{Val{:Integer}})
     end
 end
 
-N(b::Basic) = N(BasicType(b))
+N(b::BasicType{Val{:Rational}}) = Rational(N(num(b)), N(den(b))) # TODO: Implement rational_get_mpq in cwrapper.h
 N(b::BasicType{Val{:RealDouble}}) = convert(Cdouble, b)
-# TODO: Implement rational_get_mpq in cwrapper.h
-N(b::BasicType{Val{:Rational}}) = eval(parse(replace(toString(b), "/", "//")))
 N(b::BasicType{Val{:RealMPFR}}) = convert(BigFloat, b)
-N(b::BasicComplexNumber) = complex(N(real(b)), N(imag(b)))
 
-## function N(b::BasicType{Val{:Rational}})
-##     println("XXX")
-## end
-## need to test for free_symbols, if none then we need to evaluate
+N(b::BasicComplexNumber) = complex(N(real(b)), N(imag(b)))
 function N(b::BasicType)
     b = convert(Basic, b)
     fs = free_symbols(b)
     if length(fs) > 0
         throw(ArgumentError("Object can have no free symbols"))
     end
-    eval(lambdify(b))
+    out = evalf(b)
+    imag(out) == Basic(0.0) ? real(out) : out
 end
         
 
-N(a::Integer) = a
-N(a::Rational) = a
-N(a::Complex) = a
+##  Conversions SymEngine -> Julia 
+## define convert(T, x) methods leveraging N()
+convert{T <: Union{Int, BigInt,  Float64, BigFloat, Real}}(::Type{T}, x::Basic) = convert(T, BasicType(x))
+convert{T <: Union{Int, BigInt,  Float64, BigFloat, Real}}(::Type{T}, x::BasicType) = convert(T, N(x))
 
-function evalf(b::Basic, bits::Integer=53, real::Bool=false)
-    c = Basic()
-    status = ccall((:basic_evalf, libsymengine), Cint, (Ptr{Basic}, Ptr{Basic}, Culong, Cint), &c, &b, Culong(bits), Int(real))
-    if status == 0
-        return c
-    else
-        throw(ArgumentError("libsymengine has to be compiled with MPFR and MPC for eval with precision greater than 53."))
-    end
-end
+## Rational
+den(x::Basic)                     = den(BasicType(x))
+den(x::BasicType{Val{:Integer}})  = Basic(1)
+den(x::BasicType{Val{:Rational}}) = Basic(String(copy(split(SymEngine.toString(x), "/"))[2]))
+den(x::BasicComplexNumber)        = imag(x) == Basic(0) ? den(real(x)) : throw(InexactError())
+den(x::BasicType)                 = throw(InexactError())
```

Comment:
I've created symengine/symengine#1143 for this. For `a/b+c/d`, it gives `a*d+b*c` and `b*d` as `numer` and `denom`